openpype/hosts/nuke/plugins/publish/extract_thumbnail.py

```diff
@@ -166,29 +166,45 @@ def render_thumbnail(self, instance, output_name=None, **kwargs):
             previous_node = dag_node
             temporary_nodes.append(dag_node)
 
+        thumb_name = "thumbnail"
+        # only add output name and
+        # if there are more than one bake preset
+        if (
+            output_name
+            and len(instance.data.get("bakePresets", {}).keys()) > 1
+        ):
+            thumb_name = "{}_{}".format(output_name, thumb_name)
+
         # create write node
         write_node = nuke.createNode("Write")
-        file = fhead[:-1] + output_name + ".jpg"
-        name = "thumbnail"
-        path = os.path.join(staging_dir, file).replace("\\", "/")
-        instance.data["thumbnail"] = path
-        write_node["file"].setValue(path)
+        file = fhead[:-1] + thumb_name + ".jpg"
+        thumb_path = os.path.join(staging_dir, file).replace("\\", "/")
+
+        # add thumbnail to cleanup
+        if not instance.context.data.get("cleanupFullPaths"):
+            instance.context.data["cleanupFullPaths"] = []
+        instance.context.data["cleanupFullPaths"].append(thumb_path)
+
+        # make sure only one thumbnail path is set
+        # and it is existing file
+        if instance.data.get("thumbnail"):
+            _path = instance.data["thumbnail"]
+            if not os.path.isfile(_path):
+                instance.data["thumbnail"] = thumb_path
+
+        write_node["file"].setValue(thumb_path)
         write_node["file_type"].setValue("jpg")
         write_node["raw"].setValue(1)
         write_node.setInput(0, previous_node)
         temporary_nodes.append(write_node)
-        tags = ["thumbnail", "publish_on_farm"]
 
-        thumb_output_name = "thumb"
-        if output_name:
-            thumb_output_name = "{}_{}".format(output_name, thumb_output_name)
         repre = {
-            'name': name,
+            'name': thumb_name,
             'ext': "jpg",
-            "outputName": thumb_output_name,
+            "outputName": thumb_name,
             'files': file,
             "stagingDir": staging_dir,
-            "tags": tags
+            "tags": ["thumbnail", "publish_on_farm", "delete"]
```

Comment:
Wait a second, don't we still need the thumbnails on disk for the loader tool? Or what obvious thing am I missing?

Please confirm that:
> - added delete tag so it is not integrated into db as representation

Doesn't mean that new subsets don't show thumbnails anymore in the loader.
And if possible, please link me the related code where it **does** get the thumbnail from then ;)

Reply:
Also - please update the **Testing notes** description on the PR to include that test, and update the rest to what you think is the expected behavior. As I understand with this change no thumbnail representation should enter the database nor should any files exist on disk in the publish folders for the thumbnails.

Reply:
> don't we still need the thumbnails on disk for the loader tool?

We do not need it as representation.

Reply:
> We do not need it as representation.

Great - where is the Loader getting those thumbnails from?

EDIT: Let me answer this myself. Apparently versions store data:
```
'thumbnail_id': ObjectId('64f0791e24595ebe754cd8c5'),
```
And thus this is already linking by thumbnail ID.

Can we please get matching PRs for Maya + Fusion to kill thumbnails being extracted into something that "integrates" into the database?

Reply:
Do you need Thumbnails representation to be loadable? Please look into Description of the PR it should be quite clear.

Reply:
No - I don't need them loadable. 👍

Reply:
There was this`project_settings/global/publish/PreIntegrateThumbnails` to control if thumbnails should be integratated, as each host did that differently.
So by adding generic all host profile with disabled Integrate should persuade all hosts from integrating it.